src/super_gradients/training/samples/depth_estimation_sample.py

```diff
@@ -0,0 +1,35 @@
+import dataclasses
+
+import numpy as np
+
+
+__all__ = ["DepthEstimationSample"]
+
+
+@dataclasses.dataclass
+class DepthEstimationSample:
+    """
+    A dataclass representing a single depth estimation sample.
+    Contains input image and depth map.
+
+    :param image:              np.ndarray, Image of [H, W, (C if colorful)] shape.
+    :param depth_map:          Depth map of [H, W] shape.
+    """
+
+    __slots__ = ["image", "depth_map"]
+
+    image: np.ndarray
+    depth_map: np.ndarray
+
+    def __init__(self, image: np.ndarray, depth_map: np.ndarray):
```

Comment:
With depth estimation there is one important nuance that should be explicitly taken care of which is invalid areas in the target depth map.
Historically, a monocular depth estimation datasets were obtained from the stereo setup. By computing disparity map between two cameras authors came up with a depth map. However stereo matching inherently is noisy process and due to the occlusion there may be some areas that cannot be estimated. 

And to account for that often these "invalid" regions has some magic value in those pixel values. (This is dataset specific and could be anything - -1, nan, inf, 0 - whatever basically).

So we should respect this value. It is important to keep track of this value as it is also should be considered in loss and metric. So I suggested use a "nan" value and document this in both dataset and depth sample that non-finite values in depth map correspond to ignored values.

Similary a specific dataset class (NYU or any other) should take care of converting their internal depth representation format to our and mask those invalid regions as nans.

Reply:
The motivation of using "nan" is simple - any math operation `x op nan` would produce `nan` so we don't have to worry about spatial operations like resize (I'm theoretizing here actually, we need to check)

Reply:
Deliberately introducing NaNs into the code is not a good practice. 
They are sneaky and have a tendency to slowly spread. 
You multiply by NaN, and you get a NaN. You then sum and get NaN, then your gradients have NaN, and only after ten epochs everything collapse.

Reply:
I we document properly that nan correspond to invalid region and respect this value in loss and metric we should be good I think 
The alternative is to either have a validity_mask or explicit invalid_value in the sample object. Which frankly would make things more complicated as now one have to pass two objects to loss and metric classes (depth targets and valid_mask/invalid_value)

Reply:
@BloodAxe and I agreed this would be taken care of in the inheriting classes load_sample methods.
I am keepingthe code as is for now as this preprocessed version of NYUv2 already excludes nans, infs and negative values.